share/p2p/shrexeds/server.go

```diff
@@ -100,8 +100,16 @@ func (s *Server) handleStream(stream network.Stream) {
 	// we do not close the reader, so that other requests will not need to re-open the file.
 	// closing is handled by the LRU cache.
 	edsReader, err := s.store.GetCAR(ctx, hash)
-	status := p2p_pb.Status_OK
+	var status p2p_pb.Status
 	switch {
+	case err == nil:
+		defer func() {
+			if err := edsReader.Close(); err != nil {
+				log.Warnw("closing car reader", "err", err)
+			}
+		}()
+		edsReader.Close()
```

Comment:
Are closing the reader twice for a specific purpose or is this a mistake?

Reply:
Mistake. Good find, thank you!